modules/statistics/php/charts.class.inc

```diff
@@ -0,0 +1,349 @@
+     *                      order.)
+     *
+     * @return int An indexed array of data for site in the order of $labels
+     */
+    private function _getScansPerMonthData(array $data, int $siteID, array $labels)
+    {
+        $sitedata   = array_filter(
+            $data,
+            function ($row) use ($siteID) {
+                return $row['CenterID'] == $siteID;
+            }
+        );
+        $mappeddata = [];
+        foreach ($sitedata as $row) {
+            $mappeddata[$row['datelabel']] = $row['count'];
+        }
+
+        $data = [];
+        foreach ($labels as $i => $label) {
+            $data[$i] = $mappeddata[$label] ?? 0;
+        }
+        return $data;
+    }
+
+    /**
+     * Helper to handle the siterecruitment_line endpoint.
+     *
+     * @return ResponseInterface
+     */
+    private function _handleSiteLineData()
+    {
+        $DB = \NDB_Factory::singleton()->database();
+
+        $recruitmentData      = array();
+        $recruitmentStartDate = $DB->pselectOne(
+            "SELECT MIN(Date_registered) FROM candidate",
+            array()
+        );
+        $recruitmentEndDate   = $DB->pselectOne(
+            "SELECT MAX(Date_registered) FROM candidate",
+            array()
+        );
+
+        if( $recruitmentStartDate !== null && $recruitmentEndDate !== null)  {
+            $recruitmentData['labels'] = $this->_createSiteLineChartLabels(
+                new \DateTime($recruitmentStartDate),
+                new \DateTime($recruitmentEndDate)
+            );
+        }
+
+        $list_of_sites = \Utility::getSiteList(true, false);
+
+        foreach ($list_of_sites as $siteID => $siteName) {
+            $recruitmentData['datasets'][] = array(
+                "name" => $siteName,
+                "data" => $this->_getSiteLineRecruitmentData(
+                    $siteID,
+                    $recruitmentData['labels']
+                ),
+            );
+        }
+        return new \LORIS\Http\Response\JsonResponse($recruitmentData);
+    }
+
+    /**
+     * Helper to generate labels for every month between startDate and endDate.
+     *
+     * @param \DateTime $startDate The start date for the labels in YYYY-MM-DD format.
+     * @param \DateTime $endDate   The end date for the labels in YYYY-MM-DD format.
+     *
+     * @return array
+     */
+    private function _createSiteLineChartLabels(\DateTime $startDate, \DateTime $endDate) : array
+    {
+        $month = date_interval_create_from_date_string('1 month');
+        // Since we're only concerned with months, act as if $startDate
+        // is always the first of the month so that the last month doesn't
+        // get truncated.
+        $betweenDate = new \DateTime($startDate->format('Y-m-01'));
+
+        while ($betweenDate <= $endDate) {
+            $labels[] = $betweenDate->format('n-Y');
+            $betweenDate = $betweenDate->add($month);
```

Comment:
```suggestion
            $betweenDate->add($month);
```

DateTime::add() modifies the object directly.

Reply:
I think this is less clear than the way I wrote it even if it's technically true.

Reply:
Yeah I see what you're saying. I think it could also mislead people about the behaviour of the function though if they're only seeing it in this context.
 Maybe we can add what I wrote above as a comment in the code?

Reply:
they're now DateTimeImmutable and no one can be misled.